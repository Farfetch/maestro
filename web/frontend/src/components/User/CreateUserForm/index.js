import CreateUserForm from "./CreateUserForm";

export default CreateUserForm;
