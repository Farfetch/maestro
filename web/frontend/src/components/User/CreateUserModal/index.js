import CreateUserModal from "./CreateUserModal";

export default CreateUserModal;
