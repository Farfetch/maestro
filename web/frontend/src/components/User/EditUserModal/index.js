import EditUserModal from "./EditUserModal";

export default EditUserModal;
