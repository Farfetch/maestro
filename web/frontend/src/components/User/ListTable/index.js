import UserListTable from "./UserListTable";

export default UserListTable;
