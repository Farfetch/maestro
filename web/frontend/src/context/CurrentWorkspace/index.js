export {
  CurrentWorkspaceContext,
  CurrentWorkspaceContextProvider
} from "./CurrentWorkspace";
