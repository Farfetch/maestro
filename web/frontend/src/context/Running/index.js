export { RunningContext, RunningContextProvider } from "./RunningContext";
