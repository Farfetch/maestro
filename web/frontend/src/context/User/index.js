export { UserContext, UserContextProvider } from "./UserContext";
