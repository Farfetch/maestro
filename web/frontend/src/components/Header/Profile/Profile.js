import {
  AppstoreOutlined,
  LogoutOutlined,
  UserOutlined
} from "@ant-design/icons";
import { Dropdown, Menu, Select, Space } from "antd";
import { useContext } from "react";
import { Link, useLocation, useNavigate } from "react-router-dom";

import { CurrentWorkspaceContext } from "../../../context/CurrentWorkspace";
import { UserContext } from "../../../context/User";
import { userRole as userRoleModel } from "../../../lib/api/models";
import {
  historyUrl,
  homeUrl,
  logoutUrl,
  testsUrl,
  usersUrl,
  workspacesUrl
} from "../../../lib/routes";
import Avatar from "../../layout/Avatar";

const Profile = () => {
  const navigate = useNavigate();
  const { pathname } = useLocation();
  const { user, workspaces } = useContext(UserContext);
  const { currentWorkspace, setCurrentWorkspace } = useContext(
    CurrentWorkspaceContext
  );

  const onWorkspaceChange = (workspaceId) => {
    if (![testsUrl, historyUrl, homeUrl].includes(pathname))
      navigate(historyUrl);
    const workspace = workspaces.find(({ id }) => id === workspaceId);
    setCurrentWorkspace(workspace);
  };

  const menu = (
    <Menu>
      <Menu.Item key="me" disabled>
        {user.email}
      </Menu.Item>
      {user.role === userRoleModel.ADMIN ? (
        <>
          <Menu.Item key="workspaces" icon={<AppstoreOutlined />}>
            <Link to={workspacesUrl}>Workspaces</Link>
          </Menu.Item>
          <Menu.Item key="users" icon={<UserOutlined />}>
            <Link to={usersUrl}>Users</Link>
          </Menu.Item>
        </>
      ) : null}
      <Menu.Item key="logout" icon={<LogoutOutlined />}>
        <a href={logoutUrl}>Logout</a>
      </Menu.Item>
    </Menu>
  );

  return (
    <Space align="center">
      <Select
        style={{ width: 240 }}
        onChange={onWorkspaceChange}
        defaultValue={currentWorkspace.id}
      >
        {workspaces.map((workspace) => (
          <Select.Option value={workspace.id} key={workspace.id}>
            {workspace.name}
          </Select.Option>
        ))}
      </Select>
      <Dropdown
        overlay={menu}
        placement="bottomRight"
        arrow={{ pointAtCenter: true }}
      >
        <div style={{ cursor: "pointer" }}>
          <Avatar user={user} />
        </div>
      </Dropdown>
    </Space>
  );
};

export default Profile;
