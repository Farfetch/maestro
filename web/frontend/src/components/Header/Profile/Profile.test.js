import { render } from "@testing-library/react";
import React from "react";

import { UserContext } from "../../../context/User";
import Profile from "./Profile";

describe("components/Header/Profile", () => {
  test(`should render Profile component from UserContext`, async () => {
    const user = {
      id: "user-id",
      name: "Test Maestro",
      email: "test@maestro.net"
    };

    const rendered = render(
      <UserContext.Provider value={{ user }}>
        <Profile />
      </UserContext.Provider>
    );

    const profileComponent = rendered.container;
    expect(profileComponent.outerHTML).toMatchSnapshot();
  });
});
