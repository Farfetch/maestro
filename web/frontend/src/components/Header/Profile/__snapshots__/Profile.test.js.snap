// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header/Profile should render Profile component from UserContext 1`] = `"<div><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\" style=\\"height: 54px;\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 12px;\\"><span style=\\"background-color: rgb(250, 140, 22);\\" class=\\"ant-avatar ant-avatar-circle\\"><span class=\\"ant-avatar-string\\" style=\\"transform: scale(1) translateX(-50%);\\">TM</span></span></div><div class=\\"ant-space-item\\"><span class=\\"ant-typography ant-dropdown-trigger\\" style=\\"cursor: pointer;\\">test@maestro.net <span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div></div></div>"`;
